addons/mass_mailing/controllers/main.py

```diff
@@ -33,6 +35,62 @@ def mailing_unsubscribe_oneclick(self, mailing_id, email=None, res_id=None, toke
         self.mailing(mailing_id, email=email, res_id=res_id, token=token, **post)
         return Response(status=200)
 
+    @http.route('/mail/mailing/<int:mailing_id>/confirm_unsubscribe', type='http', website=True, auth='public')
+    def mailing_confirm_unsubscribe(self, mailing_id, email=None, res_id=None, token="", **post):
+        mailing = request.env['mailing.mailing'].sudo().browse(mailing_id)
+        # Check access (note that this will also raise AccessDenied if the mailing does not exist)
+        if not self._valid_unsubscribe_token(mailing_id, res_id, email, str(token)):
+            raise exceptions.AccessDenied()
+
+        unsubscribed_str = _("Are you sure you want to unsubscribe from our mailing list?")
+        # Display list name if list is public
+        if mailing.mailing_model_real == 'mailing.contact':
+            unsubscribed_lists = ', '.join(mailing_list.name or _('<i>Unnamed</i>') for mailing_list in mailing.contact_list_ids if mailing_list.is_public)
```

Comment:
I means to keep a list here (so `unsubscribed_lists` is not a string), and in the template, you can do something like
```xml

<p>
    <t t-foreach="unsubscribed_lists" t-as="mailing_list">
        <i t-if="not mailing_list">Unnamed</i>
        <span t-else="" t-out="mailing_list"/>
    </t>
</p>
```